### ISSUE TITLE
unknown value stmt at [n n n n .... n]

### ISSUE THREAD

**Issue by Reinhard-Prehofer** [uwol/cobol85parser#30]
In parsing huge programs (more than 10.000 LOCs), the parser tends to issue a lot of "**call to unknown data element []**" statements.
Could you give a hint:
- when such a message is raised?
- and how to interprete the numbers inside the [] - are these line numbers of the parsed program or internal ids or ???
- is it rather a warning or an error (msg is in black - not red)

If you need cobol samples for that issue, let me know: like mentioned above - I only observed that error pattern in "big cobol sources", so I would have to do some time consuming anonymization of the customer code before uploading 

```
call to unknown data element FUNCTIONDATE-OF-INTEGER(FUNCTIONINTEGER-OF-DATE(P0120(X-S))-1)
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4308 4296 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
unknown value stmt at [5339 5330 5312 3662 4184 4159 3632 4319 4297 3641 4308 4296 3641 3610 3602 3593 3576 3538 1200 1185 1182]
call to unknown data element x-1
```

**Comment by uwol** [comment]
The log statements are more info/warn, not error.

The message "unknown value stmt" is a bug, fixed in 8bebf21db9e6ca69ab44fc93663d357b6f58b033. The numbers inside [] are ANTLR token ids, so not to be interpreted by humans :-)

The message "call to unknown data element" is raised, when data description entries cannot be resolved on the symbol table.
* In that situation, an undefined call without a connection between the caller and callee is created. 
* This message can be caused by erroneous COBOL code or a bug in the parser. In your example it seems that function calls are not resolved correctly by the parser, so a bug in the parser.

Thanks, no source files required!

**Comment by uwol** [comment]
Fixed parser bug with function calls in 1b8c890ca51e870438a3ddfe5bbe373a4ce0d2a8. Added unit test.

Thanks!
